php/xmlrpc.php

```diff
@@ -127,32 +127,37 @@ public function getCommandsCount()
 		return(count($this->commands));
 	}
 
-	protected function makeCall()
+	protected function makeNextCall()
 	{
-	        rTorrentSettings::get()->patchDeprecatedRequest($this->commands);
 		$this->fault = false;
 		$this->content = "";
-		$cnt = count($this->commands);
+		$cnt = count($this->commands) - $this->commandOffset;
 		if($cnt>0)
 		{
 			$this->content = '<?xml version="1.0" encoding="UTF-8"?><methodCall><methodName>';
 			if($cnt==1)
 			{
-				$cmd = $this->commands[0];
+				$cmd = $this->commands[$this->commandOffset++];
 	        		$this->content .= "{$cmd->command}</methodName><params>\r\n";
 	        		foreach($cmd->params as &$prm)
 	        			$this->content .= "<param><value><{$prm->type}>{$prm->value}</{$prm->type}></value></param>\r\n";
 		        }
 			else
 			{
+				$maxContentSize = rTorrentSettings::get()->maxContentSize();
 				$this->content .= "system.multicall</methodName><params><param><value><array><data>";
-				foreach($this->commands as &$cmd)
+				for(; $this->commandOffset < count($this->commands); $this->commandOffset++)
 				{
-					$this->content .= "\r\n<value><struct><member><name>methodName</name><value><string>".
+					$cmd = $this->commands[$this->commandOffset];
+					$cmdStr = "\r\n<value><struct><member><name>methodName</name><value><string>".
 						"{$cmd->command}</string></value></member><member><name>params</name><value><array><data>";
 					foreach($cmd->params as &$prm)
-						$this->content .= "\r\n<value><{$prm->type}>{$prm->value}</{$prm->type}></value>";
-					$this->content .= "\r\n</data></array></value></member></struct></value>";
+						$cmdStr .= "\r\n<value><{$prm->type}>{$prm->value}</{$prm->type}></value>";
+					$cmdStr .= "\r\n</data></array></value></member></struct></value>";
+					if($this->commandOffset > count($this->commands) - $cnt and
```

Comment:
As we may see above, initially ```$cnt = count($this->commands) - $this->commandOffset;```
Both ```$cnt``` and ```count($this->commands)``` are unchanged in code.
At first iteration we have ```$this->commandOffset == count($this->commands) - $cnt``` i.e. condition is always false.
At second iteration this condition is always true, because we increase ```$this->commandOffset```.
I.e. this cycle will run only once.

Sorry, but do you really test your code?

Reply:
Yes, of course I tested it :smile: 
I wanted to prevent that it gets stuck in an endless loop (when testing with small maxContentSizes). Normally this will never happen since `count($cmdStr)` will never be larger than maxContentSize. But with just `if (strlen($this->content) + strlen($cmdStr) + 35 + 22 > $maxContentSize)` it could theoretically get stuck.

`if ($this->commandOffset > count($this->commands) - $cnt and strlen($this->content) + strlen($cmdStr) + 35 + 22 > $maxContentSize)` guarantees that at least one command is included in each request